kubeflow/core/kubeform_spawner.py

```diff
@@ -90,8 +99,12 @@ def extra_resource_limits(self):
 ###################################################
 # Spawner Options
 ###################################################
+cloud = os.environ.get('CLOUD_NAME')
 c.JupyterHub.spawner_class = KubeFormSpawner
-c.KubeSpawner.singleuser_image_spec = 'gcr.io/kubeflow/tensorflow-notebook'
+if cloud == 'ack':
```

Comment:
Instead of taking cloud as a parameter why not take image_spec as a parameter. This would prevent a proliferation of flags for different clouds. Instead we would just set that parameter to the desired registry path.